packages/replay/src/coreHandlers/handleBreadcrumbs.ts

```diff
@@ -7,61 +8,55 @@ import type { ReplayFrame } from '../types/replayFrame';
 import { createBreadcrumb } from '../util/createBreadcrumb';
 import { addBreadcrumbEvent } from './util/addBreadcrumbEvent';
 
-let _LAST_BREADCRUMB: null | Breadcrumb = null;
-
 type BreadcrumbWithCategory = Required<Pick<Breadcrumb, 'category'>>;
 
-function isBreadcrumbWithCategory(breadcrumb: Breadcrumb): breadcrumb is BreadcrumbWithCategory {
-  return !!breadcrumb.category;
-}
+/**
+ * Handle breadcrumbs that Sentry captures, and make sure to capture relevant breadcrumbs to Replay as well.
+ */
+export function handleBreadcrumbs(replay: ReplayContainer): void {
+  const client = getClient();
 
-export const handleScopeListener: (replay: ReplayContainer) => (scope: Scope) => void =
-  (replay: ReplayContainer) =>
-  (scope: Scope): void => {
-    if (!replay.isEnabled()) {
-      return;
-    }
+  if (!client || !client.on) {
+    return;
+  }
```

Comment:
Fine for now but this just reminded me that we can finally make `client.on` required and avoid this check whenever we use hooks

Reply:
yeah, working on this right now :D